### Problem 1:

#### Is gender independent of education level? A random sample of 395 people were surveyed and each person was asked to report the highest education level they obtained. The data that resulted from the survey is summarized in the following table:

High School Bachelors Masters Ph.d. Total

Female 60 54 46 41 201

Male 40 44 53 57 194

Total 100 98 99 98 395

### Solution:
given the data we want to know, is there a relationship between the gender of an individual and the level of education that they have obtained at 5% level of significance.

we will use Chi-Square test of independence

H0 :The two categorical variables are independent.
H1:The two categorical variables are dependent.

In [15]:
import numpy as np
import pandas as pd
import scipy.stats as stats

m_list = [40,44,53,57]
f_list = [60,54,46,41]
sample = m_list + f_list
print(sample)
print('\n')
sex =  ['Male','Male','Male','Male','Female','Female','Female','Female']
edu = ['High School', 'Bachelors', 'Masters', 'Ph.d.','High School', 'Bachelors', 'Masters', 'Ph.d.']
df= pd.DataFrame({"Sex":sex,"Edu":edu,"Sample":sample})
df
df2 = pd.crosstab(df.Sex, df.Edu,df.Sample, aggfunc="sum",margins=True)
df2
df2.columns = ["Bachelors","High School","Masters","Ph.d.","row_totals"]
df2.index = ["Female","Male","col_totals"]
df2

[40, 44, 53, 57, 60, 54, 46, 41]




Sex          Edu  Sample
0    Male  High School      40
1    Male    Bachelors      44
2    Male      Masters      53
3    Male        Ph.d.      57
4  Female  High School      60
5  Female    Bachelors      54
6  Female      Masters      46
7  Female        Ph.d.      41

Edu     Bachelors  High School  Masters  Ph.d.  All
Sex                                                
Female         54           60       46     41  201
Male           44           40       53     57  194
All            98          100       99     98  395

Bachelors  High School  Masters  Ph.d.  row_totals
Female             54           60       46     41         201
Male               44           40       53     57         194
col_totals         98          100       99     98         395

In [23]:
observed = df2.iloc[0:2,0:4]   # Get table without totals for later use
observed
expected =  np.outer(df2["row_totals"][0:2],
                     df2.loc["col_totals"][0:4]) / 395.0
expected = pd.DataFrame(expected)
expected.columns = ["Bachelors","High School","Masters","Ph.d."]
expected.index = ["Female","Male"]
expected
chi_squared_test = (((observed-expected)**2)/expected).sum().sum()

print('\n',chi_squared_test)

Bachelors  High School  Masters  Ph.d.
Female         54           60       46     41
Male           44           40       53     57

Bachelors  High School    Masters      Ph.d.
Female  49.868354    50.886076  50.377215  49.868354
Male    48.131646    49.113924  48.622785  48.131646


 8.006066246262538


In [24]:
critical_value = stats.chi2.ppf(q = 0.95, df = 3)   # Find the critical value for 95% confidence

print("Critical value")
print(critical_value)

p_value = 1 - stats.chi2.cdf(x=chi_squared_stat,  # Find the p-value
                             df=3)
print("\nP value")
print(p_value)

Critical value
7.814727903251179

P value
0.04588650089174717


#### Since 8.006 > 7.815, therefore we reject the null hypothesis and conclude that the education level depends on gender at a 5% level of significance

### Problem 2:

#### Using the following data, perform a oneway analysis of variance using α=.05. Write up the results in APA format.
#### [Group1: 51, 45, 33, 45, 67] [Group2: 23, 43, 23, 43, 45] [Group3: 56, 76, 74, 87, 56]
### Solution:

#### H0: μ1=μ2=μ3
#### H1: μ1≠μ2≠μ3

In [28]:
import scipy.stats as stats
Group1 = [51, 45, 33, 45, 67]
Group2 = [23, 43, 23, 43, 45]
Group3 = [56, 76, 74, 87, 56]
statistic, pvalue = stats.f_oneway(Group1,Group2,Group3)
print("F Statistic value {},\np-value {}".format(statistic,pvalue))

F Statistic value 9.747205503009463,
p-value 0.0030597541434430556


#### Here the p-value is 0.00305 which is < 0.05.we can reject the null hypothesis and the groups don't have the same means.

###  Problem 3:
#### Calculate F Test for given 10, 20, 30, 40, 50 and 5,10,15, 20, 25.
### Solution:

In [32]:
Group1 = [10, 20, 30, 40, 50]
Group2 = [5,10,15, 20, 25]
mean_1 = np.mean(Group1)
mean_2 = np.mean(Group2)
add1 = 0
add2 = 0
for items in Group1:
    add1 += (items - mean_1)**2
for items in Group2:
    add2 += (items - mean_2)**2
var1 = add1/(len(Group1)-1)
var2 = add2/(len(Group2)-1)

F_Test = var1/var2
F_Test

4.0